### PR TITLE
fix(llama-cpp): keep sliding-window cache bounded

### PR DESCRIPTION
**Pull request by ttupper92618**
## Summary

- make bounded sliding-window attention an explicit llama.cpp runtime invariant
- pass the shared invariant to the in-process runner and surface it in runner diagnostics
- harden the provider-overflow regression setup so its upstream transport cannot mask the downstream overflow under test

## Root cause

The in-process llama.cpp runner inherited the dependency default `swa_full=true`. Placement admitted long-context GGUF workloads using bounded sliding-window cache assumptions, while runtime expanded every sliding layer to full context. That contract mismatch could overcommit unified memory during model load.

## Impact

Long-context GGUF models now load with the same bounded SWA behavior assumed by admission. This prevents the pathological full-context KV allocation while keeping the model-card context limit unchanged.

## Validation

- `uv run basedpyright`
- `uv run ruff check`
- `nix fmt`
- `uv run pytest` — 2714 passed, 1 skipped, 222 deselected
- focused llama.cpp/admission tests — 38 passed
- provider-overflow regression repeated 5 times

### CODE REVIEW

File: src/skulk/worker/runner/llama_cpp/runner.py [hunk 1]
```diff
@@ -830,6 +837,7 @@ def _load_model(self, task: Task) -> None:
                 n_ctx=n_ctx,
                 logits_all=logits_all,
                 flash_attn=flash_attn,
+                swa_full=LLAMA_CPP_FULL_SWA_CACHE,
```
[the remark this answers is not in the document]
Comment (ttupper92618) [reply src/skulk/worker/runner/llama_cpp/runner.py:867]: Severity 4 (High): likelihood 5, impact 5, triggerability 1, complexity 1 (score 3.51). Fixed in 1b574529 by verifying that the installed `Llama` constructor explicitly declares `swa_full` before any model construction. A legacy `**kwargs`-only binding now fails closed instead of silently restoring the unbounded cache. Added a regression test for that legacy signature. Validation: basedpyright and Ruff clean; formatter changed 0 files; full pytest 2715 passed, 1 skipped, 222 deselected.
